### ISSUE TITLE
DOI download enhancements for searches on "hubs"

### ISSUE THREAD

**Issue by nickdos** [AtlasOfLivingAustralia/biocache-service#186]
The DOI downloads look to have the "original search URL" hard-coded to the server that is running the biocache-service service. 

With the various hubs all pointing to a single instance of biocache-service, the downloads need to:

- keep a copy of the original URL (full URL not just path)
- pass this into the DOI service to store
- return the full/original search URL on download page and in confirmation email (citation text)

We might also need to put the hub name in the email subject, in place of the "ALA".

**Comment by javier-molina** [comment]
If I understand correctly,  and ala-hub (and dependencies) need to tell biocache-service what the identity of the caller is (the hub name) and probably what the original search url was.

At the moment biocache-service assumes the call came from ala-hub:

https://github.com/AtlasOfLivingAustralia/biocache-service/blob/14b0b0c7df1da5503088222177c366bd54132bbd/src/main/java/au/org/ala/biocache/service/DownloadService.java#L951


The output of generateSearchUrl will eventually become the searchUrl on the DOI service, I don't think the DOI service needs any changes to accommodate this requirement.

**Comment by nickdos** [comment]
Yeah, that part needs to change in biocache-service. I'll check the downloads-plugin code to see if needs to be jacked up to pass in the full search URL or not.

**Comment by javier-molina** [comment]
downloads plugin has almost everything that is needed but currently does not pass that information to biocache.

I will work on both ends (downloads/biocache-hubs and biocache-service) to get this done, I will introduce, if not already there, a human readable property for the hub name that will be used in e-mails.

**Comment by javier-molina** [comment]
@nickdos the email content and readme file contains lines like the ones below:

When using this dataset please use the following citation:

> **Atlas of Living Australia** occurrence download at http://devt.ala.org.au/ala-hub/occurrences/search?&q=frogs&fq=sensitive%3A%22generalised%22 accessed on Fri Feb 02 15:32:39 AEDT 2018. 


Does the Atlas of Living Australia bit need to be also customised according to the hub name or this one remains static?

**Comment by nickdos** [comment]
Not sure. I'll check with Miles/

**Comment by M-Nicholls** [comment]
We should attribute the hub but don't want to lose the ALA attribution as well, lets try the example below and see if the hubs are ok with it 

Australasian Virtual Herbarium, Atlas of Living Australia occurrence download at http://devt.ala.org.au/ala-hub/occurrences/search?&q=frogs&fq=sensitive%3A%22generalised%22 accessed on Fri Feb 02 15:32:39 AEDT 2018.

**Comment by djtfmartin** [comment]
Closing historic issue. If this is still a problem, please reopen.
